sway-lib-std/src/storage/storage_vec.sw

```diff
@@ -809,7 +809,32 @@ impl<V> StorageKey<StorageVec<V>> {
     /// ```
     #[storage(write)]
     pub fn store_vec(self, vec: Vec<V>) {
-        let slice = vec.as_raw_slice();
+        let size_V_bytes = __size_of::<V>();
+
+        // Handle cases where elements are less than the size of word as pad to the size of a word
+        let slice = match size_V_bytes < 8 {
+            true => {
+                let vec_slice = vec.as_raw_slice();
+                let number_of_words = 8 * vec.len();
+                let ptr = alloc_bytes(number_of_words);
+                let mut i = 0;
+                while i < vec.len() {
+                    // Insert into raw slice as offsets of 1 word per element
+                    // (size_of_word * element)
+                    vec_slice
+                        .ptr()
+                        .add::<V>(i)
+                        .copy_bytes_to(ptr.add_uint_offset(8 * i), size_V_bytes);
+                    i += 1;
+                }
+
+                raw_slice::from_parts::<V>(ptr, number_of_words)
+            },
+            false => {
+                vec.as_raw_slice()
+            }
+        };
```

Comment:
no sense using match in place of if

Reply:
`match` is used to return two different `raw_slice`s in this case

Reply:
So?

Two branches of an if-else can return different values of the same type. These are equivalent:

```sway
match a {
    true => 1,
    false => 2,
}

if a { 1 } else { 2 }
```